diag_manager/fms_diag_file_object.F90

```diff
@@ -1096,6 +1141,46 @@ subroutine write_axis_metadata(this, diag_axis)
 
 end subroutine write_axis_metadata
 
+!< @brief Writes the field metadata for the file
+subroutine write_field_metadata(this, diag_field, diag_axis)
+  class(fmsDiagFileContainer_type), intent(inout), target :: this            !< The file object
+  class(fmsDiagField_type)        , intent(inout), target :: diag_field(:)   !<
+  class(fmsDiagAxisContainer_type), intent(in)            :: diag_axis(:)    !< Diag_axis object
+
+  class(FmsNetcdfFile_t),  pointer     :: fileobj        !< The fileobj to write to
+  class(fmsDiagFile_type), pointer     :: diag_file      !< Diag_file object to open
+
+  integer :: i           !< For do loops
+  integer :: j           !< For do loops
+  logical :: is_regional !< Flag indicating if the field is in a regional file
+  character(len=255) :: cell_measures
+
+  is_regional = this%is_regional()
+
+  diag_file => this%FMS_diag_file
+  fileobj => diag_file%fileobj
+
+  do i = 1, size(diag_file%field_ids)
```

Comment:
Should/could this be a `do concurrent`?

Reply:
this is doing io, which i think cannot be done in parallel